### ISSUE TITLE
[a11y]: ai-chat should set focus to first interactive element in full-screen mode

### ISSUE THREAD

**Issue by Daniel-Schulz-Private** [carbon-design-system/carbon-labs#883]
### Package version

0.18.0

### React version

18.3.1

### Automated testing tool and ruleset

n/a

### Description

Once the modal opened (full screen mode of ai-chat) - focus did not set to the first Logical Element within the Modal, which is "Chat Edit field".

Recommended fix: Once the modal opened - focus expected set to the first Logical Element within the Modal, which is "Chat Edit field".

User impact: Impact Keyboard /screen reader users :Without proper focus management, users may not realise a modal has opened, leading to confusion, navigation barriers, missed critical information, and a disrupted user experience.

### WCAG 2.1 Violation

WCAG Mapping: 2.4.3 Focus Order

### Reproduction/example

n/a

### Steps to reproduce

1. open ai-chat
2. enter full-screen mode
3. see that no visible element has focus set

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon-labs/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon-labs/issues) for duplicate problems
